### DISCUSSION TITLE
Is there a solution for introduce member support IntelliSense?

### DISCUSSION THREAD

**Discussion by yumikou** [metalama/Metalama#31] [Q&A] [answered]
Before using Metalama I use SourceGenerator for class add some new members , It can add a partial class by atrribute,  when i writing code,  i can access just added properties or methods by IntelliSense.

Is it possible metalama to support this feature, or can I introduce a member in a partial class like SourceGenerator?

**Comment by gfraiteur** [comment, marked as answer]
Yes, this is supported. Metalama uses source generators to implement this feature.

**Reply by yumikou** [reply]
Thanks for the reply, Later I found, Sometimes I can see this partial class in Metalama.Framework.CompilerExtensions.MetalamaSourceGenerator in Project Analyzers.
Like this:
```
/// <generated>
/// Generated by Metalama to support the code editing experience. This is NOT the code that gets executed.
/// </generated>
namespace MyLibrary
{
    partial class TestClass
    {
        public static readonly global::System.Windows.DependencyProperty CustomProperty;
    }
}
```
At this time IntelliSense works well.

But sometimes, it shows "this generator is not generating files" in Analyzers, I don't know what is causing this and I don't have any modifications.

**Reply by gfraiteur** [reply]
There may be some bugs. In this case, the best is to enable logging for the `RoslynCodeAnalysisService` process as described in https://doc.metalama.net/troubleshooting/creating-logs, and to look for an exception when the problem happens - and report it.
